src/read_dicom_file.py

```diff
@@ -4,38 +4,48 @@
 """
 
 import logging
+import os
 import pydicom
 from pydicom.errors import InvalidDicomError
 
 logger = logging.getLogger(__name__)  # Start logger
 
-
 def read_dicom_file(file_path):
-    """Read a dicom file in a fault tolerant way and return the FileDataset"""
+    """Read a DICOM file, return None if invalid or empty."""
     try:
-        # Try standard dcm read
+        # Check for empty file first
+        if os.path.getsize(file_path) == 0:
+            logger.warning("Empty file: %s", file_path)
+            return None
+            
+        # Try standard read
         dicom_file = pydicom.dcmread(file_path)
-        logging.info("Successfully read DICOM file: %s", file_path)
+
+        logger.info("Successfully read DICOM file: %s", file_path)
         return dicom_file
 
     except FileNotFoundError as e:
         logger.error("File not found: %s - %s", file_path, str(e))
         return None
 
     except InvalidDicomError as e:
-        logger.warning("File %s is not a valid DICOM file: %s",
-                       file_path, str(e))
+        logger.warning("File %s is not a valid DICOM file: %s", file_path, str(e))
```

Comment:
Given that you are logging that it's not a valid DICOM file, but you are going to try to read it with force=True, you might want to clarify that it's not a valid DICOM Part 10 file (at this point).